### ISSUE TITLE
Implement progressive pruning schedule

### ISSUE THREAD

**Issue by MaxGhenis** [PolicyEngine/L0#6]
## Description
Add support for gradually increasing sparsity during training, rather than learning it all at once.

## Motivation
Progressive pruning can lead to better final accuracy by allowing the network to adapt gradually to increasing sparsity.

## Proposed Implementation
```python
class SparsityScheduler:
    def __init__(self, initial_sparsity=0.0, final_sparsity=0.9, 
                 warmup_epochs=10, pruning_epochs=50):
        ...
    
    def step(self, epoch):
        # Update target sparsity for all L0 gates
        ...

# Usage
scheduler = SparsityScheduler(0.0, 0.9, 10, 50)
for epoch in range(100):
    scheduler.step(epoch)
    train_epoch()
```

## Features
- Linear/cosine/exponential schedules
- Per-layer sparsity targets
- Magnitude-based guidance
